src/settings/project/mod.rs

```diff
@@ -84,54 +84,50 @@ impl Manifest {
         Ok(manifest)
     }
 
-    pub fn get_target(
+    fn get_environment(
         &self,
         environment_name: Option<&str>,
-        release: bool,
-    ) -> Result<Target, failure::Error> {
-        if release && self.workers_dot_dev.is_some() {
-            failure::bail!(
-                "The --release flag is not compatible with use of the workers_dot_dev field"
-            )
-        }
-
-        if release {
-            message::warn("--release will be deprecated");
-        }
-
-        let environment = match environment_name {
-            Some(environment_name) => match &self.env {
-                Some(environment_table) => {
-                    let environment = environment_table.get(environment_name);
-                    match environment {
-                        Some(environment) => Some(environment),
-                        None => failure::bail!(format!(
-                            "{} Could not find environment with name {}",
-                            emoji::WARN,
-                            environment_name
-                        )),
-                    }
+    ) -> Result<Option<&Environment>, failure::Error> {
+        if let Some(environment_name) = environment_name {
+            if let Some(environment_table) = &self.env {
+                if let Some(environment) = environment_table.get(environment_name) {
+                    Ok(Some(environment))
+                } else {
+                    failure::bail!(format!(
+                        "{} Could not find environment with name {}",
+                        emoji::WARN,
+                        environment_name
+                    ))
                 }
-                None => failure::bail!(format!(
+            } else {
+                failure::bail!(format!(
                     "{} There are no environments specified in your wrangler.toml",
                     emoji::WARN
-                )),
-            },
-            None => None,
-        };
+                ))
+            }
+        } else {
+            Ok(None)
+        }
+    }
 
+    // TODO: when --release is deprecated, this will be much easier
+    fn negotiate_with_terrorists(
+        &self,
+        environment: Option<&Environment>,
+        release: bool,
+    ) -> Result<(Option<String>, bool), failure::Error> {
         let deprecate_warning =
             "Please specify the workers_dot_dev boolean in the top level of your wrangler.toml";
         let wdd_failure = format!(
-            "{} Your environment should only include `workers_dot_dev` or `route`",
+            "{} Your environment should only include `workers_dot_dev` or `route`. If you are trying to publish to workers.dev, add `workers_dot_dev = true`, if you are trying to publish to your own domain, add a route.",
```

Comment:
@victoriabernard92 does this look better?

Reply:
```suggestion
            "{} Your environment should only include `workers_dot_dev` or `route`. To publish to workers.dev, remove the line with route. To publish to your own domain, set `workers_dot_dev` to false ",
```

Reply:
awesome thanks!